maven-jxr-plugin/src/main/java/org/apache/maven/plugin/jxr/AbstractJxrReport.java

```diff
@@ -479,6 +552,66 @@ protected void executeReport( Locale locale )
         }
     }
 
+
+    /**
+     * Determine the templateDir to use, given javadocTemplatesVersion
+     *
+     * @return
+     */
+    private String getTemplateDir()
+    {
+        // Check if overridden
+        if ( StringUtils.isEmpty( templateDir ) )
+        {
+            if ( javadocTemplatesVersion.isAtLeast( "8" ) )
+            {
+                return "templates/jdk8";
+            }
+            else if ( javadocTemplatesVersion.isAtLeast( "7" ) )
+            {
+                return "templates/jdk7";
+            }
+            else if ( javadocTemplatesVersion.isAtLeast( "4" ) )
+            {
+                return "templates/jdk4";
+            }
+            else
+            {
+                getLog().warn( "Unsupported javadocVersion: " + javadocTemplatesVersion + ". Fallback to original" );
+                return "templates";
+            }
+        }
+        // use value specified by user
+        return templateDir;
+    }
+
+    /**
+     * Set a new value for <code>javadocTemplatesVersion</code>
+     *
+     * @throws MavenReportException if not found
+     */
+    private void setJavadocTemplatesVersion()
+        throws MavenReportException
+    {
+        JavaVersion javaVersion = JavaVersion.JAVA_SPECIFICATION_VERSION;
+
+        if ( StringUtils.isNotEmpty( javadocVersion ) )
+        {
+            try
+            {
+                javadocTemplatesVersion = JavaVersion.parse( javadocVersion );
+            }
+            catch ( NumberFormatException e )
```

Comment:
This exception will never occur (here). `JavaVersion.parse()` is a lazy evaluator. Only during comparison the value is evaluated. The evaluator is also aware of dashes, underscores and classifiers like `ea`.
Hence this method will never throw an exception.

Reply:
Ah, true :-) This is a copy'n'paste error from AbstractJavadocMojo ;-)